packages/core/strapi/lib/core-api/service/pagination.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const _ = require('lodash');
+const { has, toNumber, isUndefined, isPlainObject } = require('lodash/fp');
+
+/**
+ * Default limit values from config
+ * @return {{maxLimit: number, defaultLimit: number}}
+ */
+const getLimitConfigDefaults = () => ({
+  defaultLimit: toNumber(strapi.config.get('api.rest.defaultLimit', 25)),
+  maxLimit: toNumber(strapi.config.get('api.rest.maxLimit')) || null,
+});
+
+/**
+ * if there is max limit set and limit exceeds this number, return configured max limit
+ * @param {number} limit - limit you want to cap
+ * @param {number?} maxLimit - maxlimit used has capping
+ * @returns {number}
+ */
+const applyMaxLimit = (limit, maxLimit) => {
+  if (maxLimit && (limit === -1 || limit > maxLimit)) {
+    return maxLimit;
+  }
+
+  return limit;
+};
+
+const shouldCount = params => {
+  if (_.has(params, 'pagination.withCount')) {
+    return params.pagination.withCount === 'false' ? false : true;
```

Comment:
Can the param be `false`? In the case of a dev extend the service for instance of else. If yes, maybe we should handle the case of the boolean `false`.

Reply:
I don't understand do you mean test for a boolean type value ?

Reply:
yes exactly

Reply:
SUre I can add it